test/cypress/integration/card/payment.spec.js

```diff
@@ -205,4 +229,127 @@ describe('Standard card payment flow', () => {
       cy.get('#card-no-lbl').should('contain', 'Nid yw rhif y cerdyn yr hyd cywir')
     })
   })
+
+  describe('Secure card payment page on Apple Pay enabled browser but not Apple Pay enabled service', () => {
+    it('Should setup the payment and load the page', () => {
+      cy.task('setupStubs', createPaymentChargeStubsEnglish)
+      cy.visit(`/secure/${tokenId}`)
+
+      // 1. Charge will be created using this id as a token (GET)
+      // 2. Token will be deleted (DELETE)
+      // 3. Charge will be fetched (GET)
+      // 4. Service related to charge will be fetched (GET)
+      // 5. Charge status will be updated (PUT)
+      // 6. Client will be redirected to /card_details/:chargeId (304)
+      cy.location('pathname').should('eq', `/card_details/${chargeId}`)
+    })
+
+    it('Should enter and validate a correct card', () => {
+      cy.task('setupStubs', checkCardDetailsStubs)
+
+      cy.visit(`/card_details/${chargeId}`, {
+        onBeforeLoad: win => {
+          // Stub Payment Request API
+          if (win.PaymentRequest) {
+            // If we’re running in headed mode
+            cy.stub(win, 'PaymentRequest', getMockPaymentRequest(validPaymentRequestResponse))
+          } else {
+            // else headless
+            win.PaymentRequest = getMockPaymentRequest(validPaymentRequestResponse)
+          }
+        }
+      })
```

Comment:
if the service is not Apple Pay enable why do we need to `getMockPaymentRequest`? `getMockPaymentRequest` looks like a response from Google Pay

Reply:
🤦‍♂️ I’ve got the labels backwards, it should say Google Pay (and the Apple Pay one should say Google Pay). Will switch now

The point in this line is to add payment request API support to our cypress testing browser